src/types.rs

```diff
@@ -327,6 +329,216 @@ pub enum SortDirection {
 /// Can only be either nulls-first or nulls-last
 #[derive(Debug, PartialEq, Eq, Clone, Copy)]
 pub enum NullOrder {
-    NullsFirst,
-    NullsLast,
+    First,
+    Last,
+}
+
+/// A manifest is an immutable Avro file that lists data files or delete
+/// files, along with each file’s partition data tuple, metrics, and tracking
+/// information.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct ManifestV2 {
+    /// The table schema at the time the manifest
+    /// was written
+    pub schema: ManifestEntryV2,
+    /// ID of the schema used to write the manifest as a string
+    pub schema_id: i32,
+    /// The partition spec used to write the manifest
+    pub partition_spec: PartitionSpec,
+    /// ID of the partition spec used to write the manifest as a string
+    pub partition_spec_id: i32,
+    /// Table format version number of the manifest as a string
+    pub format_version: i32,
+    /// Type of content files tracked by the manifest: “data” or “deletes”
+    pub content: ManifestContentType,
+}
+
+/// Type of content files tracked by the manifest
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum ManifestContentType {
+    Data,
+    Deletes,
+}
+
+/// The schema of a manifest file is a struct called manifest_entry.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct ManifestEntryV2 {
+    /// field: 0
+    ///
+    /// Used to track additions and deletions.
+    pub status: ManifestEntryStatus,
+    /// field id: 1
+    ///
+    /// Snapshot id where the file was added, or deleted if status is 2.
+    /// Inherited when null.
+    pub snapshot_id: Option<i64>,
+    /// field id: 3
+    ///
+    /// Data sequence number of the file.
+    /// Inherited when null and status is 1 (added).
+    pub sequence_number: Option<i64>,
+    /// field id: 4
+    ///
+    /// File sequence number indicating when the file was added.
+    /// Inherited when null and status is 1 (added).
+    pub file_sequence_number: Option<i64>,
+    /// field id: 2
+    ///
+    /// File path, partition tuple, metrics, …
+    pub data_file: DataFileV2,
+}
+
+/// Used to track additions and deletions in ManifestEntry.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum ManifestEntryStatus {
+    /// Value: 0
+    Existing,
+    /// Value: 1
+    Added,
+    /// Value: 2
+    ///
+    /// Deletes are informational only and not used in scans.
+    Deleted,
+}
+
+/// Data file carries data file path, partition tuple, metrics, …
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct DataFileV2 {
```

Comment:
Also need a var, such as `is_partitioned` to represent if the data file is partitioned?

Reply:
I'm guess this should already be tracked in partition field (although we didn't support yet).